### PR TITLE
GetEvent and SaveEvent APIs return full superpackages even if no data filled in for a subpackage

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
GetEvent API was originally only returning the event data for filled in packages.  This update adds in the package metadata (including attributes) for sub-packages that are part of a chosen super package for an event but have no data entered.  SaveEvent also updated to return full event if successfully saved.

#### Changes
- Update SNDManager.getEventData to cycle through all subpackages of a super package even if no EventDatas are saved for the subpackage.  This is used in GetEvent API.
- Update SNDManager.updateEvent and SNDManager.createEvent to return full event if save completes successfully.
- Update Event API test driver package caching to get deep subpackages.
- Add test to Event API tests to verify empty packages returned

### CODE REVIEW

File: src/org/labkey/snd/SNDManager.java [hunk 1]
```diff
@@ -1702,19 +1703,31 @@ public void registerAttributeLookups(Container c, User u, String schema, String
...
+        {
+            SimpleFilter filter = new SimpleFilter(FieldKey.fromParts("EventDataId"), eventDataId, CompareType.EQUAL);
+            ts = new TableSelector(eventDataTable, filter, null);
 
-        EventData eventData = ts.getObject(EventData.class);
+            eventData = ts.getObject(EventData.class);
 
-        OntologyManager.getProperties(c, eventData.getObjectURI());
-        Map<String, ObjectProperty> properties = OntologyManager.getPropertyObjects(c, eventData.getObjectURI());
+            OntologyManager.getProperties(c, eventData.getObjectURI());  // TODO: What is this?
```
Comment (labkey-jeckels) [inline src/org/labkey/snd/SNDManager.java]: Can this TODO be deleted or addressed?

Comment (labkey-jeckels) [reply src/org/labkey/snd/SNDManager.java]: I don't think it should be doing anything

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]
One small cleanup suggestion based on your TODO

File: src/org/labkey/snd/SNDManager.java (hunk 1 above)
Comment (labkey-martyp) [reply src/org/labkey/snd/SNDManager.java]: Ah yes, I forgot to get rid of that.  Removed and all tested ok.

**Review by labkey-jeckels — APPROVED** [review]
